firmware/boron-ins-fsm/src/stateMachine.cpp

```diff
@@ -364,7 +364,7 @@ const char *resetReasonString(int resetReason)
 void getHeartbeat(){
 
     static unsigned long lastHeartbeatPublish = 0;
-    if((millis()-lastHeartbeatPublish) > SM_HEARTBEAT_INTERVAL || lastHeartbeatPublish == 0){
+    if((millis()-lastHeartbeatPublish) > SM_HEARTBEAT_INTERVAL || lastHeartbeatPublish == 0 || doorHeartbeatFlag){
```

Comment:
Yes for sure. I was using this flag to record when a door heartbeat was just received. Im thinking about **doorHeartbeatReceivedFlag** as "doorHeartbeatReceived" is already used to record the time since the last heartbeat was received. What do you think?

Reply:
That sounds good to me. Thanks.